OSDOCS-19149 [NETOBSERV] 1.12 Release Notes (#111255)
openshift/openshift-docs: modules/network-observability-operator-release-notes-1-12-fixed-issues.adoc

:_mod-docs-content-type: REFERENCE
[id="network-observability-operator-release-notes-1-12-fixed-issues_{context}"]
= Network Observability Operator 1.12 fixed issues

[role="_abstract"]
The Network Observability Operator 1.12 release contains several fixed issues that improve performance, system status reporting, and the user experience in the web console and CLI.

Consistent FlowCollector pipeline status::
Before this update, changes to the sampling field caused an inconsistency in the `FlowCollector` resource status. As a consequence, you could see conflicting statuses across pipeline components.
+
With this release, status reporting is made consistent across all components. As a result, the reliability of the pipeline status indicator is improved.
+
link:https://issues.redhat.com/browse/NETOBSERV-2375[NETOBSERV-2375]

Fixed dependency check for help and version commands in netobserv-cli::
Before this update, executing `help` or `version` commands in the Network Observability CLI (`netobserv-cli`) without the `yq` tool installed caused a dependency check error. As a consequence, you could not access basic utility information without installing additional tools.
+

> With this release, the dependency check is omitted for basic informational commands. As a result, `netobserv-cli` now correctly displays help and version information without requiring `yq`.

Reviewer: It's not accurate, the problem was, when running the CLI such as that:
```
oc netobserv flows --interfaces=br-ex --max-time=10s --help
```
you would expect that the "--help" flag display the help page instead of running the actual "oc netobservflows" command ; which wasn't the case, it ignored the "--help" flag when it's not provided in 1st position of the list of flags.
There's nothing about `yq` to mention here.

+
link:https://issues.redhat.com/browse/NETOBSERV-2618[NETOBSERV-2618]

Prometheus enabled by default in FlowCollector configurations::
Before this update, the default setting for Prometheus metrics was unassigned during `FlowCollector` resource creation. As a consequence, you had to manually ensure that metrics collection was active to query accurate flow data.
+
With this release, the default value for Prometheus metrics collection in the `FlowCollector` configuration is set to `true`. As a result, the deployment process is simplified and flow metrics are collected automatically.
+
link:https://issues.redhat.com/browse/NETOBSERV-2620[NETOBSERV-2620]

Fixed user interface error on Netflow Traffic page::
Before this update, navigating to the **Netflow traffic** page could trigger a minified user interface error under specific cluster conditions. As a consequence, network flow charts failed to render correctly.
+
With this release, the underlying interface rendering logic is corrected. As a result, the **Netflow traffic** page operates without errors.
+

> link:https://issues.redhat.com/browse/NETOBSERV-2643[NETOBSERV-2643]

Reviewer: This was a 4.22 specific issue, so it never was user-facing, can be removed from the release note.

Before this update, flow durations and network latencies greater than one second were improperly formatted as milliseconds (`kms`). As a consequence, donut graphs and latency metrics displayed confusing or inaccurate time designations.
+
With this release, the duration formatting function handles values above one millisecond accurately using decimal seconds. As a result, you can view precise network latency values in console charts.
+
link:https://issues.redhat.com/browse/NETOBSERV-2669[NETOBSERV-2669]

Improved FlowCollector status reporting when eBPF pods are absent::
Before this update, the `FlowCollector` resource reported a status of `Ready` even when a restrictive `nodeSelector` prevented any eBPF agent pods from deploying. As a consequence, the system status misrepresented the health of the agent layer.
+
With this release, the operator checks for a zero-pod deployment count. As a result, the `FlowCollector` resource now correctly identifies when zero eBPF pods are active, improving cluster error diagnostics.
+
link:https://issues.redhat.com/browse/NETOBSERV-2674[NETOBSERV-2674]

Optimized field exports for OpenTelemetry exporters::

> Before this update, the {OTELName} exporter processed missing or null keys as non-null data. As a consequence, unpopulated metadata fields were exported to log streams, which increased storage usage and cluttered telemetry files.

Reviewer: replace "{OTELName}" with "OpenTelemetry" I guess? (same below on line 70)

Author: We have to use attributes if there is an attribute.

Reviewer: It sounds weird, because we're not referring here to the redhat otel **product**, we're referring to the otel **protocol**.
The sentence "Before this update, the Red Hat build of OpenTelemetry exporter processed missing or null keys as non-null data." looks weird to me, if not incorrect (people may be confuse and think the fix has something to do with the redhat otel product, it does not)

Author: Ah, OK. Thank you for clarifying!

Fixed TLS volume name conflicts on OpenTelemetry exporters::
Before this update, configuring TLS certificates on OpenTelemetry exporters generated an invalid volume name format. As a consequence, the `apiserver` rejected the underlying Flow-logs Pipeline deployment specification, causing the pipeline pod to fail during initialization.
+
With this release, the operator ensures valid volume names are generated when handling TLS attributes. As a result, enabling TLS on your OpenTelemetry exporters no longer interferes with pipeline pod lifecycles.
+
link:https://issues.redhat.com/browse/NETOBSERV-2707[NETOBSERV-2707]

Kafka compression set to optional to lower eBPF agent memory usage::
Before this update, enforcing Kafka compression rules by default within certain data paths caused a significant memory overhead inside the eBPF agent layer. As a consequence, memory consumption grew unexpectedly during high-throughput runs.
+
With this release, Kafka compression is configured as an optional parameter that is disabled by default. As a result, the core memory footprint of the eBPF agent is reduced, which improves overall cluster agent stability.
+

> link:https://issues.redhat.com/browse/NETOBSERV-2733[NETOBSERV-2733]

Reviewer: this wasn't customer facing (it's for a feature introduced in the same release) so can be omitted.

// Module included in the following assemblies:
// * network_observability/network-observability-release-notes.adoc

:_mod-docs-content-type: REFERENCE
[id="network-observability-operator-release-notes-1-12-fixed-issues_{context}"]
= Network Observability Operator 1.12 fixed issues

[role="_abstract"]

> The Network Observability Operator 1.12 release contains several fixed issues that improve performance, system status reporting, and the user experience in the web console and CLI.

Reviewer 2: ```suggestion
The Network Observability Operator 1.12 release contains several fixed issues that improve performance, system status reporting, and user experience.
```
or add the {ocp} attribute in front of "web console"

Consistent FlowCollector pipeline status::
Before this update, changes to the sampling field caused an inconsistency in the `FlowCollector` resource status. As a consequence, you could see conflicting statuses across pipeline components.
+
With this release, status reporting is made consistent across all components. As a result, the reliability of the pipeline status indicator is improved.
+
link:https://issues.redhat.com/browse/NETOBSERV-2375[NETOBSERV-2375]

> Fixed help flag processing in netobserv-cli::

Reviewer 2: ```suggestion
Fixed `--help` flag processing in netobserv-cli::
```
just made me stumble a sec once i got into the text

Before this update, the `--help` flag was ignored when placed after other command flags in the Network Observability CLI. As a consequence, running commands such as `oc netobserv flows --interfaces=br-ex --max-time=10s --help` executed the flow collection instead of displaying the help page.
+
With this release, the `--help` flag is recognized regardless of its position in the command. As a result, you can now display help information by placing the `--help` flag anywhere in your command arguments.
+
link:https://issues.redhat.com/browse/NETOBSERV-2617[NETOBSERV-2617]

> Improved visibility of DNS Names warning messages::

Reviewer 2: ```suggestion
Improved visibility of DNS names warning messages::
```

Before this update, the **DNS names** graph repeatedly displayed a warning message on every refresh when running in a Prometheus-only configuration. As a consequence, the persistent warning message covered other dashboard elements.
+

> With this release, the warning message only appears during the initial data load and does not overlay other content. As a result, interface clarity is improved when navigating the dashboard.

Reviewer 2: ```suggestion
With this release, the warning message is only displayed during the initial data load and does not overlay other content. As a result, interface clarity is improved when navigating the dashboard.
```
we don't use "appears" (IBM, I think)

+
link:https://issues.redhat.com/browse/NETOBSERV-2618[NETOBSERV-2618]

Prometheus enabled by default in FlowCollector configurations::

> Before this update, the default setting for Prometheus metrics was unassigned during `FlowCollector` resource creation. As a consequence, you had to manually ensure that metrics collection was active to query accurate flow data.

Reviewer 2: ```suggestion
Before this update, the default setting for Prometheus metrics was unassigned during `FlowCollector` custom resource creation. As a consequence, you had to manually ensure that metrics collection was active to query accurate flow data.
```
a CR?

+
With this release, the default value for Prometheus metrics collection in the `FlowCollector` configuration is set to `true`. As a result, the deployment process is simplified and flow metrics are collected automatically.
+
link:https://issues.redhat.com/browse/NETOBSERV-2620[NETOBSERV-2620]

Usage examples added to CLI subcommand help text::

> Before this update, the `help` subcommands for the Network Observability CLI lacked syntax examples. As a consequence, understanding how to construct complex filtering and capture commands required additional research.

Reviewer 2: is this the same as the `--help` flag?

+
With this release, clear examples are included in the subcommand help outputs. As a result, the usability and discoverability of the CLI features are enhanced.
+
link:https://issues.redhat.com/browse/NETOBSERV-2646[NETOBSERV-2646]

Corrected latency formatting for values above one second::

> Before this update, flow durations and network latencies greater than one second were improperly formatted as milliseconds (`kms`). As a consequence, donut graphs and latency metrics displayed confusing or inaccurate time designations.

Reviewer 2: ```suggestion
Before this update, flow durations and network latencies greater than one second were improperly formatted as milliseconds. As a consequence, donut graphs and latency metrics displayed confusing or inaccurate time designations.
```

+

> With this release, the duration formatting function handles values above one millisecond accurately using decimal seconds. As a result, you can view precise network latency values in console charts.

Reviewer 2: ```suggestion
With this release, the duration formatting function handles values greater than one millisecond accurately using decimal seconds. As a result, you can view precise network latency values in console charts.
```
"above" is positional, not value

+
link:https://issues.redhat.com/browse/NETOBSERV-2669[NETOBSERV-2669]

Improved FlowCollector status reporting when eBPF pods are absent::
Before this update, the `FlowCollector` resource reported a status of `Ready` even when a restrictive `nodeSelector` prevented any eBPF agent pods from deploying. As a consequence, the system status misrepresented the health of the agent layer.
+

> With this release, the operator checks for a zero-pod deployment count. As a result, the `FlowCollector` resource now correctly identifies when zero eBPF pods are active, improving cluster error diagnostics.

Reviewer 2: ```suggestion
With this release, the Operator checks for a zero-pod deployment count. As a result, the `FlowCollector` CR now correctly identifies when zero eBPF pods are active, improving cluster error diagnostics.
```

+
link:https://issues.redhat.com/browse/NETOBSERV-2674[NETOBSERV-2674]

> Optimized field exports for OpenTelemetry exporters::

Reviewer 2: I think we have an attribute for OTEL?

Author: See https://github.com/openshift/openshift-docs/pull/111255#discussion_r3339174947

"because we're not referring here to the redhat otel product, we're referring to the otel protocol.
The sentence "Before this update, the Red Hat build of OpenTelemetry exporter processed missing or null keys as non-null data." looks weird to me, if not incorrect (people may be confuse and think the fix has something to do with the redhat otel product, it does not)"

> Before this update, the OpenTelemetry exporter processed missing or null keys as non-null data. As a consequence, unpopulated metadata fields were exported to log streams, which increased storage usage and cluttered telemetry files.

Reviewer 2: ```suggestion
Before this update, the OpenTelemetry exporter processed missing or null keys as non-null data. As a consequence, unpopulated `metadata` fields were exported to log streams, which increased storage usage and cluttered telemetry files.
```

+
With this release, the OpenTelemetry exporter filters out null or unrelated fields, exporting only keys that belong to explicitly enabled features. As a result, exported log sizes are reduced and data efficiency is improved.
+
link:https://issues.redhat.com/browse/NETOBSERV-2705[NETOBSERV-2705]

> Added sampling probability fields to IPFIX exports::

Reviewer 3: Spell out IPFIX on first reference?

Before this update, IPFIX record exports omitted per-flow sampling information. As a consequence, data exports failed to comply with the standard IPFIX specifications for `samplingProbability` usage.
+
With this release, the exporter includes sampling probability details within the IPFIX packet metadata. As a result, exported OpenTelemetry data matches industry compliance standards.
+
link:https://issues.redhat.com/browse/NETOBSERV-2706[NETOBSERV-2706]

Fixed TLS volume name conflicts on OpenTelemetry exporters::
Before this update, configuring TLS certificates on OpenTelemetry exporters generated an invalid volume name format. As a consequence, the `apiserver` rejected the underlying Flow-logs Pipeline deployment specification, causing the pipeline pod to fail during initialization.
+

> With this release, the operator ensures valid volume names are generated when handling TLS attributes. As a result, enabling TLS on your OpenTelemetry exporters no longer interferes with pipeline pod lifecycles.

Reviewer 2: ```suggestion
With this release, the Operator ensures valid volume names are generated when handling TLS attributes. As a result, enabling TLS on your OpenTelemetry exporters no longer interferes with pipeline pod lifecycles.
```

+
link:https://issues.redhat.com/browse/NETOBSERV-2707[NETOBSERV-2707]

Improved pod-to-pod flow filter rule matching for asymmetric CIDR rules::

> Before this update, the default flow filter action was not enforced when a network flow failed to match both a CIDR rule and its corresponding `peerCIDR` rule identically. As a consequence, unexpected acknowledgment-only (`ACK`) flows bypass filtering restrictions inside the pod network.

Reviewer 2: ```suggestion
Before this update, the default flow filter action was not enforced when a network flow failed to match both a CIDR rule and its corresponding `peerCIDR` rule identically. As a consequence, unexpected acknowledgment-only, `ACK`, flows bypass filtering restrictions inside the pod network.
```
IBM does not allow parens; there are a lot in this PR
